diff --git a/dist/index.mjs b/dist/index.mjs
index 27cc390481a1847fba0008ffad54c45a23a24f27..ad29b1836df715a10f2abc03879425b41e21e57c 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -26,9 +26,17 @@ var renderAsPlainText = (component, _options) => {
 // src/renderAsync.ts
 import { convert as convert2 } from "html-to-text";
 import pretty2 from "pretty";
-import { renderToReadableStream, renderToStaticMarkup as renderToStaticMarkup2 } from "react-dom/server";
+import react from 'react-dom/server';
+
+const renderToStream =
+  // Note: only available in platforms that support WebStreams
+  // https://react.dev/reference/react-dom/server/renderToString#alternatives
+  react.renderToReadableStream ||
+  // Note: only available in Node
+  react.renderToPipeableStream;
+
 async function renderToString(children) {
-  const stream = await renderToReadableStream(children);
+  const stream = await renderToStream(children);
   const html = await readableStreamToString(
     stream
   );
